be/src/exprs/array_functions.tpp

```diff
@@ -1203,4 +1203,89 @@ private:
     }
 };
 
+// Todo:support datatime/data
+template <LogicalType Type>
+class ArrayGenerate {
+public:
+    using InputColumnType = RunTimeColumnType<Type>;
+    using InputCppType = RunTimeCppType<Type>;
+    static StatusOr<ColumnPtr> process(FunctionContext* ctx, const Columns& columns) {
+        RETURN_IF_COLUMNS_ONLY_NULL(columns);
+        DCHECK(columns.size() == 3);
+
+        auto num_rows = columns[0]->size();
+
+        // compute nulls first. if any input is null, then output is null
+        NullColumnPtr nulls;
+        for (auto& column : columns) {
+            if (column->has_null()) {
+                auto nullable_column = down_cast<NullableColumn*>(column.get());
+                if (nulls == nullptr) {
+                    nulls = std::static_pointer_cast<NullColumn>(nullable_column->null_column()->clone_shared());
+                } else {
+                    ColumnHelper::or_two_filters(num_rows, nulls->get_data().data(),
+                                                 nullable_column->null_column()->get_data().data());
+                }
+            }
+        }
+
+        // collect all input columns
+        std::vector<typename InputColumnType::Ptr> input_columns;
+        for (auto& column : columns) {
+            if (column->is_nullable()) {
+                auto nullable_column = down_cast<NullableColumn*>(column.get());
+                input_columns.emplace_back(std::static_pointer_cast<InputColumnType>(nullable_column->data_column()));
+            } else if (column->is_constant()) {
```

Comment:
IMO you should optimize the performance for this most common scenario, instead of sacrifice performance for it

Reply:
you mean unpacking const column？ I fount it unnecessary and will remove it